test/matcher.cpp

```diff
@@ -113,132 +226,420 @@ TEST_CASE("Matcher throws if regex passed to `=~` or `!~` is invalid",
+			REQUIRE_FALSE(m.matches(&mock));
+
+			REQUIRE(m.parse("zero > \"unknown\""));
+			REQUIRE_FALSE(m.matches(&mock));
+
+			REQUIRE(m.parse("zero <= \"unknown\""));
+			REQUIRE(m.matches(&mock));
+
+			REQUIRE(m.parse("zero >= \"unknown\""));
+			REQUIRE(m.matches(&mock));
+
+			REQUIRE(m.parse("same_zero < \"0\""));
+			REQUIRE_FALSE(m.matches(&mock));
+
+			REQUIRE(m.parse("same_zero > \"0\""));
+			REQUIRE_FALSE(m.matches(&mock));
+
+			REQUIRE(m.parse("same_zero <= \"0\""));
+			REQUIRE(m.matches(&mock));
+
+			REQUIRE(m.parse("same_zero >= \"0\""));
+			REQUIRE(m.matches(&mock));
+		}
+	}
+
+	SECTION("Work with numbers") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
+
+		REQUIRE(m.parse("AAAA > 12344"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA > 12345"));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA >= 12345"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA < 12345"));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA <= 12345"));
+		REQUIRE(m.matches(&mock));
+	}
+
+	SECTION("Don't work with ranges") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
+
+		REQUIRE(m.parse("AAAA > 0:99999"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA < 0:99999"));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA >= 0:99999"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA <= 0:99999"));
+		REQUIRE_FALSE(m.matches(&mock));
+	}
 }
 
 TEST_CASE("Operator `between` checks if field's value is in given range",
 	"[Matcher]")
 {
-	MatcherMockMatchable mock({{"AAAA", "12345"}});
 	Matcher m;
 
-	m.parse("AAAA between 0:12345");
-	REQUIRE(m.get_parse_error() == "");
-	REQUIRE(m.matches(&mock));
+	SECTION("Doesn't work with strings") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
 
-	m.parse("AAAA between 12345:12345");
-	REQUIRE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA between \"123\""));
+		REQUIRE_FALSE(m.matches(&mock));
 
-	m.parse("AAAA between 23:12344");
-	REQUIRE_FALSE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA between \"12399\""));
+		REQUIRE_FALSE(m.matches(&mock));
+	}
 
-	m.parse("AAAA between 0");
-	REQUIRE_FALSE(m.matches(&mock));
+	SECTION("Doesn't work with numbers") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
 
-	m.parse("AAAA between 12346:12344");
-	REQUIRE(m.matches(&mock));
+		REQUIRE(m.parse("AAAA between 1"));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA between 12345"));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA between 99999"));
+		REQUIRE_FALSE(m.matches(&mock));
+	}
+
+	SECTION("Works with ranges") {
+		MatcherMockMatchable mock({{"AAAA", "12345"}});
+
+		REQUIRE(m.parse("AAAA between 0:12345"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA between 12345:12345"));
+		REQUIRE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA between 23:12344"));
+		REQUIRE_FALSE(m.matches(&mock));
+
+		REQUIRE(m.parse("AAAA between 0"));
+		REQUIRE_FALSE(m.matches(&mock));
```

Comment:
Nitpick: `0` is not a range so these 2 lines fit better in the section called `"Doesn't work with numbers"`.

Reply:
Oh, right. We already have "AAAA between 1" there, so I'll simply delete these two lines.